### PR TITLE
Use new data-form-id attribute for formId

### LINKED ISSUE #781: Update transformer to v1.43 and change access of form's id to use new data-form-id attribute
Once https://github.com/enketo/enketo-transformer/pull/105 is merged and released.

https://github.com/enketo/enketo-core/blob/master/src/js/form.js#L160

### PR DESCRIPTION
**Pull request by lognaturel**
Closes #781

See https://github.com/enketo/enketo-transformer/issues/100

### CODE REVIEW

**Review by eyelidlessness — APPROVED** [review]

File: src/js/form.js [hunk 1]
```diff
@@ -158,7 +158,7 @@ Form.prototype = {
      * @type {string}
      */
     get id() {
-        return this.view.html.id;
+        return this.view.html.attr( 'data-form-id' );
```
Comment (eyelidlessness) [inline src/js/form.js:161]: This should probably be `this.view.html.dataset.formId`

**Review by eyelidlessness — CHANGES_REQUESTED** [review]

File: src/js/form.js (hunk 1 above)
Comment (eyelidlessness) [reply src/js/form.js:161]: And this should have a test

Comment (lognaturel) [reply src/js/form.js]: Yiiiikes! I am really not to be trusted. 🙏🙏🙏

**Review by eyelidlessness — APPROVED** [review]

**Review by MartijnR — APPROVED** [review]
Thanks!

**Pull request merged**